doc-content/enterprise-only/openshift/ldap_rhsso.adoc

```diff
@@ -0,0 +1,95 @@
+// This include file requires the following attribute settings:
```

Comment:
General comment about includes of snippets (like this one). You and I have discussed before, and the only alternative I can think of at the moment, instead of defining snippets of info and including it, is to have a single module for all three "Deploying a single-authoring/managed/immutable environment/kie server" section that heavily uses ifevals to determine what is rendered: one of the three types of deployment based on the/a context defined in the assembly. Then the rh-sso snippet would be obsolete because you can simply put the content there, ifeval'd accordingly. This is something I've done in the migration doc and although it can make the module messy, it actually makes building in the future cleaner and somehow is conceptually less foggy, imo - i.e., better to keep the mess in individual impacted files than multiple messy files and snippet includes.

Again, for the sake of time, you can ignore this comment and just consider it for future reference. Snippet includes can be avoided in most cases by heavy ifevals, so it's a question of opinion, modular adherence, and tooling support down the road.

Reply:
Honestly, I can not even start imagining how I could manage maintenance with the approach you suggest. The sso-ldap include is about the maximum level of conditioning that I can as yet manage - it's already becoming too complicated. 

The suggested module would be much longer, would have many more conditionals, and also would need a local attribute anyway - defined in main.adoc no less - because in some cases it would be included twice with different settings.

In theory it might be cleaner, but it is definitely beyond my ability to handle.

Reply:
(But yes, this might be an opinion thing. I find snippets cleaner than complicated conditionals, which led to having three snippets in the openshift space already).

Reply:
Things would be a bit simpler if asciidoctor allowed attribute definition in the middle of a bulleted list. I will submit this as a bug but I don't expect anything to change in the short or middle term.

Reply:
As discussed: leaving this alone for now with the understanding that it might clash with Aura requirements as that comes in. I'll try to raise with Aura once we're out of release crunch.